### ISSUE TITLE
warning: no case matching constant switch condition     '0'

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#43]
drivers/block/drbd/* has tons of warnings like:
```sh
In file included from drivers/block/drbd/drbd_actlog.c:30:
In file included from drivers/block/drbd/drbd_int.h:48:
In file included from ./include/linux/drbd_genl_api.h:54:
In file included from ./include/linux/genl_magic_struct.h:236:
./include/linux/drbd_genl.h:317:1: warning: no case matching constant switch condition
      '0'
GENL_struct(DRBD_NLA_NOTIFICATION_HEADER, 23, drbd_notification_header,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/linux/genl_magic_struct.h:220:10: note: expanded from macro 'GENL_struct'
        switch (0) {                                                    \
                ^
```

Without looking at the source code for more context `switch(0)` already looks hyper-wrong (`if (false)`), but maybe there's another way to accomplish whatever they're using that construct for.  Probably needs `CONFIG_BLK_DEV_DRBD=y`.

**Comment by nickdesaulniers** [comment]
Seems to have been added in commit ec2c35ac1ea28 ("drbd: prepare the transition from connector to genetlink")

**Comment by bulwahn** [comment]
Philipp Klocke resend Arnd Bergmann's patch that was not merged inn 2017 to resolve this issue here:
https://lore.kernel.org/r/4ffc2255-2ae2-60d4-703e-151979efe7ab@gmx.de/

**Comment by bulwahn** [comment]
Sure, go ahead and do so. My student Philipp is currently on vacation, and the patch is authored by Arnd Bergmann anyway.

**Comment by nathanchance** [comment]
Patch resent: https://lore.kernel.org/lkml/20181002171447.10765-1-natechancellor@gmail.com/

**Comment by nickdesaulniers** [comment]
Thanks for the resend @nathanchance .

@bulwahn sounds like you're a teacher?  Anything we can do to help involve students in kernel or llvm development?

**Comment by bulwahn** [comment]
@nickdesaulniers I am actually just a software engineer, probably at the moment more in the role of a software project manager. I am interested in Linux kernel development, and try to offer students the possibility to work on the Linux kernel, paying them to read code, find issues, use code analysis and write patches.

I think your blog motivated to get started around some activities with the kernel and clang. This github issue tracker fits very well to keep track of who is currently involved with which issue and collects the state of discussion quite well. Thanks for this effort!

Other than keeping up this good work, I do currently not see much other things to be done. I might eventually report on my experience with running Facebook's Infer static analyser on the kernel, and another supporting person is planned to look at the clang-tidy results in bit more in detail, which I can also report on once we have some actual deeper insights.

**Comment by nickdesaulniers** [comment]
> paying them to read code, find issues, use code analysis and write patches.

Out of pocket? :zap: 

> I think your blog motivated

Well then writing it wasn't a complete wash. :haircut_man: I haven't blogged too much about the Clang built kernel stuff, probably should.

> I might eventually report on my experience with running Facebook's Infer static analyser on the kernel

This whole effort started because I wanted to run Clang's static analyzer on the kernel, and started picking up other dependencies until building the kernel outright became the predominate blocker to so much.

> Other than keeping up this good work

I appreciate everyone's help on this, including yours.

**Comment by nathanchance** [comment]
Patch v2 with Lars' suggested change: https://lore.kernel.org/lkml/20181004200912.32348-1-natechancellor@gmail.com/

**Comment by dileks** [comment]
@nathanchance Suggestions for a v3:

In [1]:
Subject: s/pointless switch statment/pointless switch stat*e*ment (missing "e" in statement)

If you send out a v3 you can add Arnd's "Fixes:" tag, too.
From [2]:
Fixes: ec2c35ac1ea2 ("drbd: prepare the transition from connector to genetlink")

[1] https://lore.kernel.org/r/20181004200912.32348-1-natechancellor@gmail.com/
[2] https://lore.kernel.org/r/20170201165532.2605534-1-arnd@arndb.de/

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a52c5a16cf19d8a85831bb1b915a221dd4ffae3c
